src/__experimental__/components/form-field/form-field.component.js

```diff
@@ -14,29 +16,47 @@ const FormField = ({
   labelInline,
   labelWidth,
   size
-}) => (
-  <FormFieldStyle>
-    { label && (
-      <Label
-        align={ labelAlign }
-        help={ labelHelp }
-        inline={ labelInline }
-        inputSize={ size }
-        labelWidth={ labelWidth }
-      >
-        { label }
-      </Label>
-    ) }
-
-    { children }
-
-    { fieldHelp && (
-      <FieldHelp labelInline={ labelInline } labelWidth={ labelWidth }>
-        { fieldHelp }
-      </FieldHelp>
-    ) }
-  </FormFieldStyle>
-);
+}) => {
+  const [hasHover, setHasHover] = useState(false),
+      [hasFocus, setHasFocus] = useState(false);
+
+  const contextForFormField = () => {
+    return {
+      hasFocus,
+      hasHover,
+      onFocus: () => setHasFocus(true),
+      onBlur: () => setHasFocus(false),
+      onMouseOver: () => setHasHover(true),
+      onMouseOut: () => setHasHover(false)
+    };
+  };
+
+  return (
+    <FormFieldStyle>
+      <FormFieldContext.Provider value={ contextForFormField() }>
```

Comment:
Does it make more sense to have the `FormFieldStyle` inside of the `FormFieldContext`? Not sure if there'd be a situation where the former would want to consume the `context`?

Reply:
I followed the pattern used in `input-presentation.component.js` for this - this way keeps the style components as simple as possible (if they _do_ need to use any of this info we can pass it down as props rather than subscribing the Style component to the context)